### ISSUE TITLE
[java] Cannot start the Java LS in debug mode

### ISSUE THREAD

**Issue by kittaakos** [theia-ide/theia#2326]
I got this when starting the server with `DEBUG_MODE = true`.
```
ERROR: JDWP Transport dt_socket failed to initialize, TRANSPORT_INIT(510)
JDWP exit error AGENT_ERROR_TRANSPORT_INIT(197): No transports initialized [debugInit.c:750]
```

**Comment by kittaakos** [comment]
One has to change the order of the arguments. It seems, the debug mode has never worked before:
```ts
        if (debug) {
            args.push('-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=1044');
        }
        args.push.apply(args, __spread([
            '-Declipse.application=org.eclipse.jdt.ls.core.id1',
            '-Dosgi.bundles.defaultStartLevel=4',
            '-Declipse.product=org.eclipse.jdt.ls.core.product'
        ]));
        if (debug) {
            args.push('-Dlog.level=ALL');
        }
```

**Comment by kittaakos** [comment]
Closed via #1845.
